ibis/backends/duckdb/__init__.py

```diff
@@ -745,8 +719,23 @@ def _register_in_memory_table(self, op: ops.InMemoryTable) -> None:
             )
 
         name = op.name
-        table = op.data.to_pyarrow(schema)
-        self.read_in_memory(table, table_name=name)
+        # If op.data is backed by a Pandas dataframe, to_pyarrow
+        # can take several seconds for larger dataframes
+        if isinstance(op.data, DataFrameProxy):
+            table = op.data.to_frame()
+        else:
+            table = op.data.to_pyarrow(schema)
```

Comment:
I think we should follow this up with a `to_native`, since we have other backends that can work with both (not for compute, avoiding the copy going to one or the other is the point):

- duckdb
- polars
- bigquery
- clickhouse (not yet implemented)